astropy/table/serialize.py

```diff
@@ -150,6 +154,32 @@ def _construct_mixin_from_obj_attrs_and_info(obj_attrs, info):
     return mixin
 
 
+class _TableLite(OrderedDict):
```

Comment:
Might it be an idea to use `TableColumns` instead? It would mean either adding a few properties to it, or slightly reworking `_construct_mixin_from_columns`.

Reply:
The methods/properties needed here are a strictly orthogonal set to `TableColumns`.  The only thing in common is the `OrderedDict` base and the concept of containing columns.  So I think it makes sense to not clutter up the core `TableColumns` class for this purpose.

Reply:
OK, let's leave it for now. In the end we'd probably want `Table` to be fast enough that it doesn't matter...

Reply:
When Table is changed to allow a mixture of MaskedColumn and Column in the same table, then we can probably revert this.  Performance wasn't the driver, but the issue in the comment for this class.